packages/payment-processor/test/payment/erc20-transferable-receivable.test.ts

```diff
@@ -0,0 +1,187 @@
+      const request = deepCopy(validRequest);
+      request.currencyInfo.network = '';
+      await expect(payErc20TransferableReceivableRequest(request, wallet)).rejects.toThrowError(
+        'Payment currency must have a network',
+      );
+    });
+
+    it('should throw an error if request has no extension', async () => {
+      const request = deepCopy(validRequest);
+      request.extensions = [] as any;
+
+      await expect(payErc20TransferableReceivableRequest(request, wallet)).rejects.toThrowError(
+        'PaymentNetwork not found',
+      );
+    });
+
+    it('should consider override parameters', async () => {
+      const spy = jest.fn();
+      const originalSendTransaction = wallet.sendTransaction.bind(wallet);
+      wallet.sendTransaction = spy;
+      await payErc20TransferableReceivableRequest(validRequest, wallet, undefined, undefined, {
+        gasPrice: '20000000000',
+      });
+      expect(spy).toHaveBeenCalledWith({
+        data: '0x314ee2d90000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000006400000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c5fdf4076b8f3a5357c5e395ab970b5b54098fef0000000000000000000000000000000000000000000000000000000000000008d23709a91f6b135f000000000000000000000000000000000000000000000000',
+        gasPrice: '20000000000',
+        to: '0xF426505ac145abE033fE77C666840063757Be9cd',
+        value: 0,
+      });
+      wallet.sendTransaction = originalSendTransaction;
+    });
+
+    it('should pay an ERC20 transferable receivable request with fees', async () => {
+      // first approve the contract
+      const approvalTx = await approveErc20(validRequest, wallet);
+      const approvalTxReceipt = await approvalTx.wait(1);
+
+      expect(approvalTxReceipt.status).toBe(1);
+      expect(approvalTx.hash).not.toBeUndefined();
+
+      // get the balance to compare after payment
+      const balanceEthBefore = await wallet.getBalance();
+      const balanceErc20Before = await getErc20Balance(validRequest, payeeWallet.address, provider);
+
+      const tx = await payErc20TransferableReceivableRequest(validRequest, wallet, 1, 0, {
+        gasLimit: BigNumber.from('20000000'),
+      });
+
+      const confirmedTx = await tx.wait(1);
+
+      const balanceEthAfter = await wallet.getBalance();
+      const balanceErc20After = await getErc20Balance(validRequest, payeeWallet.address, provider);
+
+      expect(confirmedTx.status).toBe(1);
+      expect(tx.hash).not.toBeUndefined();
+
+      expect(balanceEthAfter.lte(balanceEthBefore)).toBeTruthy(); // 'ETH balance should be lower'
+
+      // ERC20 balance should be lower
+      expect(
+        BigNumber.from(balanceErc20After).eq(BigNumber.from(balanceErc20Before).add(1)),
+      ).toBeTruthy();
+    });
+  });
+});
```

Comment:
Please add tests that expect transferable-receivable-specific validation errors

* https://github.com/00labs/requestNetwork/blob/pr/mliu/1033/packages/payment-processor/src/payment/utils.ts#L359-L360
* https://github.com/00labs/requestNetwork/blob/pr/mliu/1033/packages/payment-processor/src/payment/utils.ts#L371-L372

Reply:
The latter one actually is unreachable since we added in the validate for fee proxy logic which checks the same thing, will remove.